fastlane_core/lib/fastlane_core/project.rb

```diff
@@ -172,6 +172,72 @@ def app_name
       return "App" # default value
     end
 
+    def build_for?
+      return :ios_app if ios_app?
+      return :ios_framework if ios_framework?
+      return :ios_library if ios_library?
+      return :mac_app if mac_app?
+      return :mac_library if mac_library?
+      return :mac_framework if mac_framework?
+      return :command_line_tool if command_line_tool?
+    end
+
+    def dynamic_library?
+      return true if build_settings(key: "PRODUCT_TYPE") == "com.apple.product-type.library.dynamic"
```

Comment:
One more thing for all of these `?` methods... would you mind just returning the boolean expression rather than `true if boolean-expression`?